src/passes/DeadCodeElimination.cpp

```diff
@@ -321,84 +322,62 @@ struct DeadCodeElimination : public WalkerPass<PostWalker<DeadCodeElimination>>
     }
   }
 
-  void visitSetLocal(SetLocal* curr) {
-    if (isUnreachable(curr->value)) {
-      replaceCurrent(curr->value);
+  // Append the reachable operands of the current node to a block, and replace
+  // it with the block
+  void BlockifyReachableOperands(std::vector<Expression*> list, WasmType type) {
+    for (size_t i = 0; i < list.size(); ++i) {
```

Comment:
but i is only ever used as an index into a std::vector, never to interact with any `Index`-related part of the IR. and we aren't handling overflow anyway?